File: python-client-sdk/src/sdk/models/shared/runprojectrequestbody.py
from __future__ import annotations
import dataclasses
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Any, Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class RunProjectRequestBody:
    r"""RunProjectRequestBody
    Request body format for triggering a project run
    """
    
    dry_run: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dryRun'), 'exclude': lambda f: f is None }})
    input_params: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inputParams'), 'exclude': lambda f: f is None }})
    update_cache: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updateCache'), 'exclude': lambda f: f is None }})
    

File: java-client-sdk/lib/src/main/java/models/shared/RunProjectRequestBody.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * RunProjectRequestBody
 * Request body format for triggering a project run
**/
public class RunProjectRequestBody {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dryRun")
    public Boolean dryRun;
    public RunProjectRequestBody withDryRun(Boolean dryRun) {
        this.dryRun = dryRun;
        return this;
    }
    
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("inputParams")
    public java.util.Map<String, Object> inputParams;
    public RunProjectRequestBody withInputParams(java.util.Map<String, Object> inputParams) {
        this.inputParams = inputParams;
        return this;
    }
    
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("updateCache")
    public Boolean updateCache;
    public RunProjectRequestBody withUpdateCache(Boolean updateCache) {
        this.updateCache = updateCache;
        return this;
    }
    
}
